src/portalbackend/PortalBackend.DBAccess/Repositories/CompanyInvitationRepository.cs

```diff
@@ -104,4 +104,14 @@ public void AttachAndModifyCompanyInvitation(Guid invitationId, Action<CompanyIn
             .Where(x => x.Id == invitationId)
             .Select(x => x.ServiceAccountUserId)
             .SingleOrDefaultAsync();
+
+    public Task<(Guid? ApplicationId, Guid? CompanyId)> GetInvitationCompanyAndApplicationId(Guid invitationId) =>
+        _context.CompanyInvitations
+            .Where(x => x.Id == invitationId)
+            .Select(x => new ValueTuple<Guid?, Guid?>(
+
+                x.ApplicationId,
+                x.Application!.CompanyId
+            ))
```

Comment:
You can change the return type to directly return `CompanyInvitationDetails`

```suggestion
    public Task< CompanyInvitationDetails?> GetInvitationCompanyAndApplicationId(Guid invitationId) =>
        _context.CompanyInvitations
            .Where(x => x.Id == invitationId)
            .Select(x => new CompanyInvitationDetails(
                x.ApplicationId,
                x.Application!.CompanyId
            ))
```

Reply:
@Phil91 : Actually this Record is created in Admin Service so i don't think we should reference the Admin service in repository

Reply:
or shall i create a model in DB.Access library?